src/order/closure.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Bhavik Mehta
+-/
+import order.basic
+import order.preorder_hom
+import tactic.monotonicity
+
+/-!
+# Closure operators on a partial order
+
+We define (bundled) closure operators on a partial order as an monotone (increasing), extensive
+(inflationary) and idempotent function.
+We define closed elements for the operator as elements which are fixed by it.
+-/
```

Comment:
Do you mind adding the wikipedia link as a reference here (along with any other sources that might be helpful)?

Reply:
Added! I can't think of any other helpful references off the top of my head